### PR TITLE
Document instructions for minmal dask-core install

### PR DESCRIPTION
**Pull request by weiji14**
This pull request's core purpose is to add in instructions for installing dask-core which has less dependency bloat than the default `conda install dask`. A use-case for example is  for developers building docker images that don't want a overly huge image size (see e.g. [here](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#avoid-installing-unnecessary-packages)), or for those who would like to manage the optional dependencies themselves.

Aware of https://github.com/dask/dask/pull/2336 which removes confusion of two channels (default and conda-forge), but seeing as dask-core downloads overwhelming come from the conda-forge channel (>30k vs < 100 see https://anaconda.org/search?q=dask-core), I thought I'd add back in the instructions for conda-forge, but add in 'recommended' in brackets next to default channel so that new users won't be too confused. Open to recommendations on what's better.

External references:
https://pypi.python.org/pypi/dask
https://github.com/conda/conda/pull/4982 (note new optional dependency functionality from conda 4.4.0 onwards)
https://github.com/conda-forge/dask-feedstock/issues/22

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`
- [ ] Fully documented, including `docs/source/changelog.rst` for all changes
      and one of the `docs/source/*-api.rst` files for new API

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -12,15 +12,19 @@ Conda
 
 Dask is installed by default in `Anaconda <https://www.anaconda.com/download/>`_::
 
-You can update Dask using `conda <https://www.anaconda.com/download/>`_::
+You can update Dask using the `conda <https://www.anaconda.com/download/>`_: command:
 
-    conda install dask
+*   ``conda install dask``               : Installs dask from the default channel (recommended)
+*   ``conda install -c conda-forge dask``: Installs dask from `conda-forge <https://conda-forge.github.io/>`_.
```
Comment (mrocklin) [inline docs/source/install.rst]: I suspect that having both commands here will confuse new users, who will think that they need to run both commands. I'm inclined to stick with the "packages are maintained on both channels" message from before.

There is also some cost to adding a new channel here if the user doesn't use conda-forge (or more likely just doesn't understand the concept of channels).  The dependency packages like numpy are still slightly different on the two channels, so this will likely cause a significant download.  My experience is that people who know about conda-forge are also well-enough informed to add the `-c conda-forge` while users who don't know about conda-forge are generally not well served with the added choice.

File: docs/source/install.rst [hunk 2]
```diff
@@ -12,15 +12,19 @@ Conda
...
 
-This installs Dask and all common dependencies, including Pandas and NumPy.
+This installs Dask and **all** common dependencies, including Pandas and NumPy.
```
Comment (mrocklin) [inline docs/source/install.rst]: I like the emphasis here, especially given your experience with the bloated install.

File: docs/source/install.rst [hunk 3]
```diff
@@ -12,15 +12,19 @@ Conda
...
 
-Dask packages are maintained both on the default channel and on and
-`conda-forge <https://conda-forge.github.io/>`_.
+Optionally, you can obtain a no frills dask installation using:
```
Comment (mrocklin) [inline docs/source/install.rst]: I recommend that we change the "no frills" idiom to "minimal" for non-native-english speakers

File: docs/source/install.rst [hunk 4]
```diff
@@ -12,15 +12,19 @@ Conda
...
 
+*   ``conda install dask-core``                : Installs dask-core from the default channel (recommended)
```
Comment (mrocklin) [inline docs/source/install.rst]: How about "Installs only the dask library".  I suspect that referring to `dask-core` as a thing will confuse people a bit.

File: docs/source/install.rst [hunk 5]
```diff
@@ -12,15 +12,19 @@ Conda
...
+*   ``conda install -c conda-forge dask-core`` : Installs dask-core from `conda-forge <https://conda-forge.github.io/>`_.
+    
+This will install a minimal set of dependencies required to run dask, similar to (but not exactly the same as) ``pip install dask`` below. See https://github.com/conda-forge/dask-feedstock/issues/22 for more information on the large number of dependencies `conda install dask` would pull.
```
Comment (mrocklin) [inline docs/source/install.rst]: I don't think that we should link to github here.  If there is a need to describe the situation more fully then we should do so here as the central source of information.

File: docs/source/install.rst (hunk 1 above)
Comment (weiji14) [reply docs/source/install.rst]: Agreed, will keep it simple as it is. I do like the format xarray uses (see [here](https://github.com/pydata/xarray/blob/6ccbdfd065e135bc58c409ff8fb6b2f4739fb4d5/doc/installing.rst#instructions)), but again, that's probably for advanced use cases only.

File: docs/source/install.rst (hunk 2 above)
Comment (weiji14) [reply docs/source/install.rst]: Cool, keeping it then :smile:

File: docs/source/install.rst (hunk 3 above)
Comment (weiji14) [reply docs/source/install.rst]: Strangely enough, I did think about that. I'll change it to minimal then.

Also, reverting the removed 'Dask packages ... on default and on ... conda-forge' line. There was actually an extra 'and' in there so I'll remove that.

File: docs/source/install.rst (hunk 5 above)
Comment (weiji14) [reply docs/source/install.rst]: Sure, people should be able to find out more through a simple git blame/web search. It did take me a while to track down what the situation was (I went from `pip install dask` to `conda install dask` and found myself with a lot of unnecessary bloat), but I guess this PR change will solve that. I'll remove the reference link.

File: docs/source/install.rst (hunk 4 above)
Comment (weiji14) [reply docs/source/install.rst]: Good spotting, that was just a quick copy-edit-paste I did from the previous lines. I reckon we won't even need the description to be honest, dask-core seems to be self-explanatory. There is already an emphasis that 'dask' is the whole suite and 'dask-core' is well, the core/bare minimum. I'll just stick with ``conda install dask-core`` (with no description).

File: docs/source/install.rst [hunk 6]
```diff
@@ -12,15 +12,19 @@ Conda
 
 Dask is installed by default in `Anaconda <https://www.anaconda.com/download/>`_::
 
-You can update Dask using `conda <https://www.anaconda.com/download/>`_::
+You can update Dask using the `conda <https://www.anaconda.com/download/>`_: command:
 
-    conda install dask
+   ``conda install dask``
```
Comment (mrocklin) [inline docs/source/install.rst]: Why this change?  Why not stick to standard code block with the double colon?

File: docs/source/install.rst [hunk 7]
```diff
@@ -12,15 +12,19 @@ Conda
...
 
-Dask packages are maintained both on the default channel and on and
-`conda-forge <https://conda-forge.github.io/>`_.
+Dask packages are maintained both on the default channel and on `conda-forge <https://conda-forge.github.io/>`_.
 
+Optionally, you can obtain a minimal dask installation using:
+
+   ``conda install dask-core``
+    
+This will install a very minimal set of dependencies required to run dask, similar to (but not exactly the same as) ``pip install dask`` below.
```
Comment (mrocklin) [inline docs/source/install.rst]: "very minimal" -> "minimal"

Minimal is already superlative, can't be more minimal than minimal :)

File: docs/source/install.rst [hunk 8]
```diff
@@ -12,15 +12,19 @@ Conda
...
+   ``conda install dask``
 
-This installs Dask and all common dependencies, including Pandas and NumPy.
+This installs Dask and **all** common dependencies, including Pandas and NumPy.
 
-Dask packages are maintained both on the default channel and on and
-`conda-forge <https://conda-forge.github.io/>`_.
+Dask packages are maintained both on the default channel and on `conda-forge <https://conda-forge.github.io/>`_.
 
+Optionally, you can obtain a minimal dask installation using:
```
Comment (mrocklin) [inline docs/source/install.rst]: Perhaps "using the following command::"

File: docs/source/install.rst (hunk 6 above)
Comment (weiji14) [reply docs/source/install.rst]: Ok, change reverted. Sorry, a bit new to the rst format, more used to md markdown (probably not a good excuse though 😝 ).

File: docs/source/install.rst (hunk 8 above)
Comment (weiji14) [reply docs/source/install.rst]: Done, plus I've taken out an extra ":" that snuck in somewhere else.

File: docs/source/install.rst (hunk 7 above)
Comment (weiji14) [reply docs/source/install.rst]: Pardon my English, I guess I'm overemphasizing again. I'll revert that.

File: docs/source/install.rst (hunk 8 above)
Comment (mrocklin) [reply docs/source/install.rst]: The extra `:` is intentional.  http://docutils.sourceforge.net/docs/user/rst/quickref.html#literal-blocks

File: docs/source/install.rst [hunk 9]
```diff
@@ -12,19 +12,19 @@ Conda
 
 Dask is installed by default in `Anaconda <https://www.anaconda.com/download/>`_::
 
-You can update Dask using the `conda <https://www.anaconda.com/download/>`_: command:
```
Comment (weiji14) [inline docs/source/install.rst]: Sorry @mrocklin , was referring to this ":" here.

File: docs/source/install.rst (hunk 8 above)
Comment (weiji14) [reply docs/source/install.rst]: See my new comment below, it's not that colon 😳 But thanks for the rst link, I'll definitely need to practice that more 😃

File: docs/source/install.rst (hunk 9 above)
Comment (mrocklin) [reply docs/source/install.rst]: Ah, great

**Comment by jakirkham** [comment]
Anything else pending for this PR?

**Comment by mrocklin** [comment]
Yes.  This looks great.  I apologize for letting this slip.  Merging

Thanks @weiji14 !

**Comment by jakirkham** [comment]
np. Just happened by it through an xref and it looked nice. Thanks all. :)

**Comment by weiji14** [comment]
:grin:
